### PR TITLE
refactor: Use length caching in for-loops

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

**Comment by gregor** [comment]
Technical question out of curiosity: Why doesn't the linter complain? The length we cache is never reassinged. It should be `const` but is a `let`. It should trip of the linter but apparently does not. Why?

File: app/script/util/Crypto.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ z.util.Crypto = {
       let hash = uint32.toUint32(0);
       const key = string.toLowerCase();
 
-      for (let index = 0; index <= key.length - 1; index++) {
```
Comment (AndyLnd) [inline app/script/util/Crypto.js:32]: ```index < key.length```

File: app/script/util/ArrayUtil.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ z.util.ArrayUtil = {
       return undefined;
     }
 
-    for (let index = nextIndex; index <= array.length; index++) {
```
Comment (AndyLnd) [inline app/script/util/ArrayUtil.js:57]: ```index < array.length```
otherwise the last will be out of bounds and therefore ```undefined```

**Comment by bennycode** [comment]
Closed in favor of other array performance optimizations like:
- https://github.com/wireapp/wire-webapp/pull/4307

**Pull request closed without merging**